components/TextFields/src/MDCTextInputControllerBase.m

```diff
@@ -1450,10 +1456,32 @@ - (UIEdgeInsets)textInsets:(UIEdgeInsets)defaultInsets {
   // .bottom = underlineOffset + MDCTextInputControllerBaseDefaultPadding
   // Legacy default has an additional padding here but this version does not.
   textInsets.bottom = underlineOffset + MDCTextInputControllerBaseDefaultPadding;
-
   return textInsets;
 }
 
+// This method is used to calculate the actual number of lines for the label at the time.
+// The logic was sourced from
+// https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/TextLayout/Tasks/CountLines.html.
+- (NSUInteger)calculatedNumberOfLinesForLabel:(UILabel *)label {
```

Comment:
Consider making this a static method in order to reduce the likelihood of reliance on state.